Add SUMO rules mentions & Improve AAQ step 3

denyshon opened the pull request:
### Changes
- Add SUMO rules mentions when creating/editing a question, an answer, a forum thread, a forum post, a KB forum thread or a KB forum post ([#2196](https://github.com/mozilla/sumo/issues/2196)).
- In AAQ step 3, replace "Subject:" with "Summarize your question" and add help text for this field.
- In AAQ step 3, display the Troubleshooting Info help text above the Share Data button, right after the field name.
- In AAQ step 3, add the "tb_version" extra field ([#2536](https://github.com/mozilla/sumo/issues/2536)).
- In AAQ step 3, autofill the OS version field for Thunderbird.
- Fix the issue with the Share Data button not working when editing a question ([#1642](https://github.com/mozilla/sumo/issues/1642)).
- Do not display the Common responses button when editing a question.
- Improve accessibility.
- Use "Windows 10/11" when there is not enough info to identify the exact Windows NT 10.0 version ([#2700](https://github.com/mozilla/sumo/issues/2700)).
- Add l10n comments.
- Improve i18n, internationalize some alt strings
- Move repeated Helpful Tip code to a macro
- Remove an unused variable from forms.py

Resolves [#2196](https://github.com/mozilla/sumo/issues/2196), [#2536](https://github.com/mozilla/sumo/issues/2536), [#1642](https://github.com/mozilla/sumo/issues/1642), [#2700](https://github.com/mozilla/sumo/issues/2700).

### Screenshots

<details>
<summary>AAQ step 3 (Firefox): [click to expand]</summary>
<img width="2526" height="4784" alt="image_2025-12-08_17-48-27" src="https://github.com/user-attachments/assets/5032f512-31b5-4032-82af-39333cc5f4d9" />
</details>

<details>
<summary>AAQ step 3 (Firefox) after clicking the Share Data button, in case we failed to pull the data: [click to expand]</summary>
<img width="2526" height="4928" alt="image_2025-12-08_17-50-11" src="https://github.com/user-attachments/assets/468d963a-a446-479f-a1b8-76429750dc9f" />
</details>

<details>
<summary>AAQ step 3 (Thunderbird): [click to expand]</summary>
<img width="2526" height="4816" alt="image_2025-12-08_17-52-38" src="https://github.com/user-attachments/assets/b955c8cb-4d35-4bc2-af92-e0a2fd78b7bd" />
</details>

<details>
<summary>An example of how the editor looks on the updated surfaces (except for questions): [click to expand]</summary>
<img width="2526" height="3828" alt="image_2025-12-08_17-58-57" src="https://github.com/user-attachments/assets/efc114f9-6364-429f-99cc-cc0c1e548cfc" />
</details>

_Note: Some icons in the screenshots may be missing because they were taken on a test setup._

### TODO [admin]
Once this PR is merged, the following changes have to be made via the admin console:
- "tb_version" needs to be added to the Thunderbird & Thunderbird for Android AAQ Configurations' extra fields (for the "Thunderbird version" field to be displayed).
- (optional, but recommended) The "Windows 10/11" tag needs to be created (for it to be applied when we can't determine whether a user is on Windows 10 or Windows 11).

kelimuttu commented:
Hi Dennys, 

As always, thanks for being proactive. Really appreciate your initiative to work on these improvements. Generally, I like the direction of these changes. I think the first one is especially needed for our forum. It'll be great if we can seek feedback from someone with UX expertise to help advise with the copy and the text placing itself. 

Also, I'm not sure if it's a good idea to mix bunch of changes into 1 PR. At least, I'm pretty sure the other improvements on the AAQ page will need to be discussed on the product level. So I won't comment much on that and will defer them to platform team.

denyshon commented:
@wsmwk has requested a placeholder for the "tb_version" field. Here's how the Thunderbird AAQ step 3 looks now:
<details>
<summary>[click to expland]</summary>
<img width="2526" height="4816" alt="image_2025-12-09_10-16-15" src="https://github.com/user-attachments/assets/92430e7a-1b06-4cd9-9c7b-290c34cff1ac" />
</details>

plwt commented:
> @plwt that's not important at all, but since we (correctly) do not display this button when creating a question, it makes perfect sense not to display it when editing a question too. A common response can't be the question content anyway, so...

Sounds good to me - glad it is included, no problems here. :-)

denyshon commented:
Thank you, everyone, for your feedback! I've incorporated all the suggestions, except for the newline (that one is minor and we'll be able to add it later if necessary).
To avoid merging-related issues, I've rebased this PR in #7159. The changes are the same, so I'll close this one in favor of clean #7159. Time to get a review from the kitsune team now!
